packages/flutter/lib/src/services/text_input.dart

```diff
@@ -661,14 +660,14 @@ class TextEditingValue {
     return TextEditingValue(
       text: encoded['text'] as String,
       selection: TextSelection(
-        baseOffset: encoded['selectionBase'] as int ?? -1,
-        extentOffset: encoded['selectionExtent'] as int ?? -1,
+        baseOffset: encoded['selectionBase'] as int? ?? -1,
+        extentOffset: encoded['selectionExtent'] as int? ?? -1,
         affinity: _toTextAffinity(encoded['selectionAffinity'] as String) ?? TextAffinity.downstream,
```

Comment:
The `?? TextAffinity.downstream` is strange since `_toTextAffinity` effectively can't return null (see comment above).

Reply:
Maybe `encoded['selectionAffinity']` can actually be null?

Reply:
No idea :-( Let's run tests and see...

Reply:
Actually this is null on web plateform and is the cause of #62724.